pkg/controller/eks/nodegroup/controller.go

```diff
@@ -219,3 +229,70 @@ func (t *tagger) Initialize(ctx context.Context, mg resource.Managed) error {
 	}
 	return errors.Wrap(t.kube.Update(ctx, cr), errKubeUpdateFailed)
 }
+
+func versionHasChanged(cr *manualv1alpha1.NodeGroup, ng *awseks.DescribeNodegroupOutput) bool {
+	if ng.Nodegroup.LaunchTemplate != nil && cr.Spec.ForProvider.LaunchTemplate != nil {
+		if !reflect.DeepEqual(ng.Nodegroup.LaunchTemplate.Version, cr.Spec.ForProvider.LaunchTemplate.Version) {
+			return true
+		}
+	}
+	return !reflect.DeepEqual(ng.Nodegroup.Version, cr.Spec.ForProvider.Version)
+}
+
+func generateInput(ctx context.Context, e *external, cr *manualv1alpha1.NodeGroup, ng *awseks.DescribeNodegroupOutput) (*awseks.UpdateNodegroupVersionInput, error) {
+	input := &awseks.UpdateNodegroupVersionInput{
+		ClusterName:   &cr.Spec.ForProvider.ClusterName,
+		NodegroupName: awsclient.String(meta.GetExternalName(cr)),
+	}
+
+	var customAMI bool
+	if ng.Nodegroup.LaunchTemplate != nil {
+		crVersion, err := strconv.ParseInt(*cr.Spec.ForProvider.LaunchTemplate.Version, 10, 64)
+		if err != nil {
+			return &awseks.UpdateNodegroupVersionInput{}, awsclient.Wrap(err, "cannot convert template version")
+		}
+		describeInput := &awsec2.DescribeLaunchTemplateVersionsInput{
+			LaunchTemplateName: cr.Spec.ForProvider.LaunchTemplate.Name,
+			Versions:           []string{*cr.Spec.ForProvider.LaunchTemplate.Version},
+		}
+		ltp, err := e.ec2Client.DescribeLaunchTemplateVersions(ctx, describeInput)
+		if err != nil {
+			return &awseks.UpdateNodegroupVersionInput{}, awsclient.Wrap(err, "cannot describe launchtemplate version")
+		}
+		for _, v := range ltp.LaunchTemplateVersions {
+			if *v.VersionNumber == crVersion {
+				if v.LaunchTemplateData.ImageId != nil {
+					customAMI = true
+				}
+				break
+			}
+		}
```

Comment:
I don't think you need to query the LaunchTemplate to check for a custom AMI. If the NodeGroup was deployed with a LaunchTemplate that specifies an AMI the `amiType` attribute of the NodeGroup will be set to `CUSTOM`.

Checking the LaunchTemplate itself might be necessary if it would be possible to change from "managed AMI" to "custom AMI". Therefore I ran a test - luckily it is not possible:

```
An error occurred (InvalidParameterException) when calling the UpdateNodegroupVersion operation: You cannot specify an image id within the launch template, since your nodegroup is configured to use an EKS optimized AMI.
```

Therefore, it could be inferred that a custom AMI is being used from the value of `DescribeNodegroupOutput.Nodegroup.amiType`.

Reply:
@baldorch, that is a better way of doing it!